src/main/java/org/spongepowered/api/item/inventory/Inventory.java

```diff
@@ -30,20 +30,22 @@
 import org.spongepowered.api.data.property.PropertyHolder;
 import org.spongepowered.api.event.item.inventory.InteractInventoryEvent;
 import org.spongepowered.api.item.ItemType;
+import org.spongepowered.api.item.inventory.property.SlotIndex;
 import org.spongepowered.api.item.inventory.query.QueryOperation;
 import org.spongepowered.api.item.inventory.transaction.InventoryTransactionResult;
+import org.spongepowered.api.item.inventory.type.ViewableInventory;
 import org.spongepowered.api.plugin.PluginContainer;
 import org.spongepowered.api.util.ResettableBuilder;
 
 import java.util.Collection;
+import java.util.List;
 import java.util.Optional;
-import java.util.Queue;
 import java.util.function.Consumer;
 
 /**
  * Base interface for queryable inventories.
  */
-public interface Inventory extends Iterable<Inventory>, Nameable, PropertyHolder {
+public interface Inventory extends Nameable, PropertyHolder {
```

Comment:
I want to hear the arguments for this change. It was at the heart of the Inventory API from the get go to not impose any presumption of ordering and, instead, leave that to the OrderedInventory interface (as that is more a Container concept).

Reply:
The reason is simple:
All Inventories that we currently have implementations for are accessible by index and getting a Slot of an Inventory by index was a lot of work.
You either had to check&cast to OrderedInventory, which only works for some inventories.
Or query for it but...
When you query for SlotIndex.of(0) you wont always get a single Slot instead all Slots that have index 0 recursivly. That includes the first Slot of every Row/Column.
When you query for OrderedInventory you wont always get one OrderedInventory instead EmptyInventory/OrderedInventory/multiple OrderedInventory.

Iterable was moved to `children()` instead. Because a lot of ppl. found it confusing.